### PR TITLE
Anti ban evasion.

### PR DESCRIPTION
**Pull request by ghost**
Ports the anti ban evasion script from CM, no idea if it'll work but it's worth a shot.

Made by Kyle Spier-Swenson

### CODE REVIEW

File: code/AntiEvade.dm [hunk 1]
```diff
@@ -0,0 +1,19 @@
+//Anti-Evader 1.0
+//SPECIAL THANKS TO Kyle Spier-Swenson FOR MAKING MY DREAM A REALITY.
+//The code is like, 99% his.
+
+var/list/client/clientcidcheck = list()
```
Comment (ike709) [inline code/AntiEvade.dm:5]: This should be a proper global list.

File: code/AntiEvade.dm [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+/client/New()
```
Comment (ike709) [inline code/AntiEvade.dm:7]: Put this code in the actual client/New instead of duping the proc.

File: code/AntiEvade.dm [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+	. = ..()
+	if (isnull(clientcidcheck[ckey]))
+		clientcidcheck[ckey] = computer_id
+		src << link("byond://[world.internet_address]:[world.port]")
+	else
+		if (clientcidcheck[ckey] != computer_id)
+			to_chat(src, "Please remove wsock32.dll from c:/program files/byond/bin and reconnect.")
+			message_admins("[clientcidcheck[ckey]] may be using Evasion Tools")
+			log_game("[clientcidcheck[ckey]] may be using Evasion Tools")
+			del(src)
```
Comment (ike709) [inline code/AntiEvade.dm:17]: Never call del.

**Comment by ike709** [comment]
Pretty sure the existing CID randomizer checker essentially does the same thing.

**Comment by qwertyquerty** [comment]
This already exists

**Comment by Crossedfall** [comment]
https://github.com/BeeStation/BeeStation-Hornet/blob/1e7c0ddeb371b0aa356d671b72e028edc6788363/code/modules/client/client_procs.dm#L736 for reference

**Pull request closed without merging**